packages/core/src/RenderingEngine/helpers/getOrCreateCanvas.ts

```diff
@@ -54,5 +61,32 @@ export default function getOrCreateCanvas(
   const internalDiv =
     element.querySelector(viewportElement) || createViewportElement(element);
 
-  return internalDiv.querySelector(canvasSelector) || createCanvas(internalDiv);
+  const canvas = (internalDiv.querySelector(canvasSelector) ||
+    createCanvas(internalDiv)) as HTMLCanvasElement;
+  // Fit the canvas into the div
+  const rect = internalDiv.getBoundingClientRect();
+  const devicePixelRatio = window.devicePixelRatio || 1;
+
+  // The left/top can be fractional physical pixels, round UP to the nearest
+  // physical pixel on the left/top hand edge.
+  const left = Math.ceil(rect.x * devicePixelRatio) - rect.x * devicePixelRatio;
+  const top = Math.ceil(rect.y * devicePixelRatio) - rect.y * devicePixelRatio;
+  // The width/height is the number of physical pixels which fit into the
+  // remaining space without any fractional pixels left over
+  // The floor is because we don't want any fractional pixels left over.
+  const width = Math.floor(rect.width * devicePixelRatio - left);
```

Comment:
@sedghi  - as you indicated in the review, this calculation can leave a single pixel wide wrong colour.  I'm going to address your comment by having the viewable canvas just larger than the div, and cut it off so that one gets a fractional extra pixel rather than missing a fractional pixel.